lib/components/primitive-components/Group/Group_doInitialSchematicLayoutMatchPack.ts

```diff
@@ -119,6 +120,14 @@ function convertTreeToInputProblem(
         // If schFacingDirection is set, don't allow rotation
         availableRotations = [0]
       }
+      // Check for components with _getFacingDirection method (e.g., PinHeader with defaults)
+      if (
+        (component as PinHeader)._getFacingDirection &&
+        typeof (component as PinHeader)._getFacingDirection === "function"
```

Comment:
this is really incorrect

Reply:
```tsx
if ((component as PinHeader)?._getFacingDirection()) {
```

Reply:
or even better

```tsx
if (component._getSchFacingDirection()) {
```

then define in the PrimitiveComponent `_getFacingSchFacingDirection() { return null }`